include/caffe/layer.hpp

```diff
@@ -316,6 +317,21 @@ class Layer {
     param_propagate_down_[param_id] = value;
   }
 
+  bool IsForwardPassed() const {
```

Comment:
You can't, of course. Instead, you'd need to set `forward_passed_` in layer code, presumably at the end of `Forward_*`.